src/Adapter/Product/AdminProductWrapper.php

```diff
@@ -487,19 +487,38 @@ public function processProductCustomization($product, $data)
 
         $shopList = Shop::getContextListShopID();
 
+        /** Update the customization fields to be deleted in the next step */
+        $updateQuery = 'UPDATE ' . _DB_PREFIX_ . 'customization_field cf
+            INNER JOIN ' . _DB_PREFIX_ . 'customized_data cd ON cd.index <> cf.id_customization_field 
+            SET cf.is_deleted = 0 
+            WHERE 
+            cf.is_deleted = 1 ';
+        \Db::getInstance()->execute($updateQuery);
+
+        /** update deleted customization */
+        $updateQuery = 'UPDATE ' . _DB_PREFIX_ . 'customization_field cf
+            INNER JOIN ' . _DB_PREFIX_ . 'customized_data cd ON cd.index = cf.id_customization_field 
+            SET cf.is_deleted = 1 
+            WHERE 
+            cf.is_deleted = 0 
+            AND cf.id_product = ' . (int)$product->id;
+        $updateQuery .= empty($customization_ids) ? '' : ' AND cf.id_customization_field NOT IN (' . implode(",", $customization_ids) . ')';
```

Comment:
If your first condition return nothing, maybe you can use just a if for the good case isn't ?

```php
if (!empty($customization_ids)) {
    $updateQuery .= ' AND cf.`id_customization_field` NOT IN (' . implode(',', array_map('intval', $customization_ids)) . ')';
}
```
something like this ?